### PR TITLE
Copy link feature

### LINKED ISSUE #51: [FEATURE] Add copy link button
image

- Please add hover state as other buttons
- Clicking copy link button should copy link to users clipboard
- After succesfull click user should see Tooltip says 'Copied to clipboard!'. 
Tooltip component: https://react-bootstrap.github.io/components/overlays/#tooltips:~:text=Copy-,Tooltips,-%23

### PR DESCRIPTION
**Pull request by elyessAR**
## Fixes Issue

closes #51 

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by furkanmutlu** [comment]
image

hmm still looks like you have 3 commits. And also please close previous PR. Try to update your main branch with mine. Then create a branch, do your changes and push. 

Or you can reset back your changes on your current branch and rebase it to main. Then make your changes and push.

**Comment by elyessAR** [comment]
Screenshot
@furkanmutlu  I did exactly that, updated my main with yours, however the ahead commits didn't get removed, the only way I removed them is by creating a new branch from updated main  -> deleted the old one ->rename the new one to main however doing that require a new commit, I'm not sure but I guess my main will always be ahead a couple of commits even after rebase (because that's will require a new commit)  unless I refork and reclone the whole repo. correct me if I'm wrong

**Comment by elyessAR** [comment]
> > I did exactly that, updated my main with yours, however the ahead commits didn't get removed, the only way I removed them is by creating a new branch from updated main -> deleted the old one ->rename the new one to main however doing that require a new commit, I'm not sure but I guess my main will always be ahead a couple of commits even after rebase (because that's will require a new commit) unless I refork and reclone the whole repo. correct me if I'm wrong
> 
> hmm.. First of all click Sync fork in your fork. Then go to your local main branch and do a `git pull`. Then go to your branch(copy-link-feature) try to do `git reset [commit id of the head]`. To see commit id for HEAD, do `git log` and copy id of the HEAd. Basically it will reset your commits to have a branch same with main. Then your commits will be converted to changes. Do `git stash` to stash your changes. Then just in case do `git rebase main`. Then bring your changes back with `git stash apply`. Then make your 1 commit. and force push `git push --force-lease`.

This worked! but only when I didn't rebase to main, I just did git reset to latest head commit, commited my changes and push.
I will try to do the same think to my main to fix it, Thank you!

File: frontend/src/components/Gallery/Overlay/Overlay.jsx [hunk 1]
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
       >
         <FontAwesomeIcon icon={faExpandAlt} inverse/>
       </button>
+      <button 
+        className="copy-button"
+        ref={target}
+        onClick={() => {
+          setShowToolTip(!showToolTip);
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Overlay/Overlay.jsx:46]: please move this function up

File: frontend/src/components/Gallery/Overlay/Overlay.jsx [hunk 2]
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
...
+          onCopy();
+          setTimeout(function(){setShowToolTip(false);},2000);
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Overlay/Overlay.jsx:48]:
```suggestion
          setTimeout(() => setShowTooltip(false), 2000);
```

File: frontend/src/components/Gallery/Overlay/Overlay.jsx [hunk 3]
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
...
+        }}
+      >
+        <FontAwesomeIcon icon={faLink} inverse/>
+      </button>
+        <CopyButtonOverlay target={target.current} show={showToolTip} placement="top" rootCloseEvent="mousedown">
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Overlay/Overlay.jsx:53]: Seems like there is spacing issue here, try to format the document.

File: frontend/src/components/Gallery/Overlay/Overlay.jsx [hunk 4]
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
...
+         {(props) => (
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Overlay/Overlay.jsx:54]: Do you need to pass this props ? Seems like there is no need for it.

File: frontend/src/components/Gallery/Overlay/Overlay.jsx [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
 
 const Overlay = ({
   liked,
   onLike, 
-  onExpand, 
+  onExpand,
+  onCopy,
 }) => {
+  const [showToolTip, setShowToolTip] = useState(false);
+  const target = useRef(null);
```
Comment (furkanmutlu) [inline frontend/src/components/Gallery/Overlay/Overlay.jsx:19]:
```suggestion
  const target = useRef();
```

**Review by furkanmutlu — CHANGES_REQUESTED** [review]

**Review by furkanmutlu — APPROVED** [review]
Just added few changes. Seems like there is an issue with your linting. I also removed settiomeout to hide tooltip. Instead I added onmouse out event to do it.

**Pull request merged**